lib/impress.jstp.js

```diff
@@ -7,23 +7,34 @@ impress.jstp.mixImpress = true;
 impress.jstp.createServer = (
   config // Impress server worker configuration
 ) => {
-  let transport;
-
-  if (config.transport === 'tcp' || config.transport === 'tls') {
-    transport = api.jstp.tcp;
-  } else if (config.transport === 'ws' || config.transport === 'wss') {
-    transport = api.jstp.ws;
-  } else {
-    return null;
+  const transportModule =
+    config.transport === 'tcp' || config.transport === 'ipc' ?
+      'net' :
+      config.transport;
+  const transport = api.jstp[transportModule];
+  if (!transport) {
+    return;
   }
 
-  let apps, auth;
   if (config.name === 'master') {
-    apps = { impress };
-    auth = impress.cloud.startSession;
+    impress.version = impress.version || '1.0.0';
+    config.applications = [ impress ];
+    config.authPolicy = (
+      connection, application, strategy, [nodeId, cloudAccessKey], callback
+    ) => {
+      impress.cloud.startSession(
```

Comment:
You should also change this method because now it passes generated `sessionId` to the callback as a second argument, while `username` (which in this case will probably be just `nodeId`) must be passed instead.